lfs/transfer_queue.go

```diff
@@ -133,6 +141,14 @@ func newTransferQueue(dir transfer.Direction, meter progress.Meter, dryRun bool)
 		rc:            newRetryCounter(config.Config),
 	}
 
+	for _, opt := range options {
+		opt(q)
+	}
+
+	if q.meter == nil {
+		q.meter = progress.Noop()
```

Comment:
It'll remove the conditional, but it'll also create a dummy struct that we never need when the `lfs.WithProgress()` option is applied.